--- a/src/hotspot/os/posix/signals_posix.cpp	Wed Oct  7 17:21:51 2020
+++ b/src/hotspot/os/posix/signals_posix.cpp	Sat Oct 10 16:46:21 2020
@@ -445,6 +445,10 @@
 extern "C" JNIEXPORT int JVM_handle_aix_signal(int signo, siginfo_t* siginfo,
                                                void* ucontext,
                                                int abort_if_unrecognized);
+#elif defined(SOLARIS)
+extern "C" JNIEXPORT int JVM_handle_solaris_signal(int signo, siginfo_t* siginfo,
+                                               void* ucontext,
+                                               int abort_if_unrecognized);
 #else
 extern "C" JNIEXPORT int JVM_handle_linux_signal(int signo, siginfo_t* siginfo,
                                                void* ucontext,
@@ -513,6 +517,8 @@
   JVM_handle_bsd_signal(sig, info, uc, true);
 #elif defined(AIX)
   JVM_handle_aix_signal(sig, info, uc, true);
+#elif defined(SOLARIS)
+  JVM_handle_solaris_signal(sig, info, uc, true);
 #else
   JVM_handle_linux_signal(sig, info, uc, true);
 #endif
@@ -1314,6 +1320,8 @@
    return os::Bsd::ucontext_get_pc(ctx);
 #elif defined(LINUX)
    return os::Linux::ucontext_get_pc(ctx);
+#elif defined(SOLARIS)
+   return os::Solaris::ucontext_get_pc(ctx);
 #else
    VMError::report_and_die("unimplemented ucontext_get_pc");
 #endif
@@ -1326,6 +1334,8 @@
    os::Bsd::ucontext_set_pc(ctx, pc);
 #elif defined(LINUX)
    os::Linux::ucontext_set_pc(ctx, pc);
+#elif defined(SOLARIS)
+   os::Solaris::ucontext_set_pc(ctx, pc);
 #else
    VMError::report_and_die("unimplemented ucontext_set_pc");
 #endif
