--- a/src/hotspot/os/posix/signals_posix.cpp	Wed Oct  7 17:21:51 2020
+++ b/src/hotspot/os/posix/signals_posix.cpp	Sat Oct 10 16:46:21 2020
@@ -505,6 +505,8 @@
 #define JVM_HANDLE_XXX_SIGNAL JVM_handle_aix_signal
 #elif defined(LINUX)
 #define JVM_HANDLE_XXX_SIGNAL JVM_handle_linux_signal
+#elif defined(SOLARIS)
+#define JVM_HANDLE_XXX_SIGNAL JVM_handle_solaris_signal
 #else
 #error who are you?
 #endif
@@ -1381,6 +1387,8 @@
    return os::Bsd::ucontext_get_pc(ctx);
 #elif defined(LINUX)
    return os::Linux::ucontext_get_pc(ctx);
+#elif defined(SOLARIS)
+   return os::Solaris::ucontext_get_pc(ctx);
 #else
    VMError::report_and_die("unimplemented ucontext_get_pc");
 #endif
@@ -1393,6 +1401,8 @@
    os::Bsd::ucontext_set_pc(ctx, pc);
 #elif defined(LINUX)
    os::Linux::ucontext_set_pc(ctx, pc);
+#elif defined(SOLARIS)
+   os::Solaris::ucontext_set_pc(ctx, pc);
 #else
    VMError::report_and_die("unimplemented ucontext_set_pc");
 #endif
