--- a/adbc_driver_manager/current_arch.h	Wed Jul  2 05:25:30 2025
+++ b/adbc_driver_manager/current_arch.h	Tue Jul  8 20:34:54 2025
@@ -53,6 +53,8 @@
   static const std::string platform = "openbsd";
 #elif defined(__linux__)
   static const std::string platform = "linux";
+#elif defined(__sun__)
+  static const std::string platform = "illumos";
 #else
   static const std::string platform = "unknown";
 #endif
@@ -107,7 +109,7 @@
 // but apparently it doesn't define __USE_GNU inside of features.h
 // while gcc DOES define that.
 // see https://stackoverflow.com/questions/58177815/how-to-actually-detect-musl-libc
-#if defined(_WIN32) || defined(__APPLE__) || defined(__FreeBSD__)
+#if defined(_WIN32) || defined(__APPLE__) || defined(__FreeBSD__) || defined (__sun__)
 #else
 #if !defined(_GNU_SOURCE)
 #define _GNU_SOURCE
