--- a/cpp/src/arrow/vendored/xxhash/xxhash.h	Wed Mar 13 13:37:59 2024
+++ b/cpp/src/arrow/vendored/xxhash/xxhash.h	Mon Apr 22 16:13:05 2024
@@ -1696,7 +1696,7 @@
 
 #ifndef XXH_NO_XXH3
 
-#if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 201112L) /* >= C11 */
+#if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 201112L) && !defined(__cplusplus) /* >= C11 */
 #  define XXH_ALIGN(n)      _Alignas(n)
 #elif defined(__cplusplus) && (__cplusplus >= 201103L) /* >= C++11 */
 /* In C++ alignas() is a keyword */
@@ -2509,7 +2509,7 @@
 
 /* note: use after variable declarations */
 #ifndef XXH_STATIC_ASSERT
-#  if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 201112L)    /* C11 */
+#  if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 201112L) && !defined(__cplusplus)    /* C11 */
 #    define XXH_STATIC_ASSERT_WITH_MESSAGE(c,m) do { _Static_assert((c),m); } while(0)
 #  elif defined(__cplusplus) && (__cplusplus >= 201103L)            /* C++11 */
 #    define XXH_STATIC_ASSERT_WITH_MESSAGE(c,m) do { static_assert((c),m); } while(0)
--- a/cpp/src/arrow/util/value_parsing.h	Wed Mar 13 13:37:59 2024
+++ b/cpp/src/arrow/util/value_parsing.h	Mon Apr 22 16:15:57 2024
@@ -824,7 +824,7 @@
   if (!ignore_time_in_day) {
     secs += (std::chrono::hours(result.tm_hour) + std::chrono::minutes(result.tm_min) +
              std::chrono::seconds(result.tm_sec));
-#if !defined(_WIN32) && !defined(_AIX)
+#if !defined(_WIN32) && !defined(_AIX) && !defined(__sun)
     secs -= std::chrono::seconds(result.tm_gmtoff);
 #endif
   }
--- a/cpp/src/arrow/util/float16.h	Mon Apr 22 16:20:23 2024
+++ b/cpp/src/arrow/util/float16.h	Mon Apr 22 16:20:41 2024
@@ -74,7 +74,7 @@
   constexpr uint16_t bits() const { return bits_; }
 
   /// \brief Return true if the value is negative (sign bit is set)
-  constexpr bool signbit() const { return (bits_ & 0x8000) != 0; }
+  constexpr bool arsignbit() const { return (bits_ & 0x8000) != 0; }
 
   /// \brief Return true if the value is NaN
   constexpr bool is_nan() const { return (bits_ & 0x7fff) > 0x7c00; }
@@ -161,8 +161,8 @@
     return (lhs.bits() == rhs.bits()) || (lhs.is_zero() && rhs.is_zero());
   }
   static constexpr bool CompareLt(Float16 lhs, Float16 rhs) {
-    if (lhs.signbit()) {
-      if (rhs.signbit()) {
+    if (lhs.arsignbit()) {
+      if (rhs.arsignbit()) {
         // Both are negative
         return lhs.bits() > rhs.bits();
       } else {
@@ -169,7 +169,7 @@
         // Handle +/-0
         return !lhs.is_zero() || rhs.bits() != 0;
       }
-    } else if (rhs.signbit()) {
+    } else if (rhs.arsignbit()) {
       return false;
     } else {
       // Both are positive
--- a/cpp/src/parquet/statistics.cc	Tue Apr 16 15:48:56 2024
+++ b/cpp/src/parquet/statistics.cc	Mon Apr 22 20:21:56 2024
@@ -389,10 +389,10 @@
     return ::std::nullopt;
   }
 
-  if (min.is_zero() && !min.signbit()) {
+  if (min.is_zero() && !min.arsignbit()) {
     min_flba = FLBA{Float16Constants::negative_zero()};
   }
-  if (max.is_zero() && max.signbit()) {
+  if (max.is_zero() && max.arsignbit()) {
     max_flba = FLBA{Float16Constants::positive_zero()};
   }
 
--- a/cpp/src/arrow/compute/kernels/aggregate_var_std_internal.h	Tue Apr 22 18:59:37 2025
+++ b/cpp/src/arrow/compute/kernels/aggregate_var_std_internal.h	Wed May 28 15:54:55 2025
@@ -88,10 +88,10 @@
     double result;
     // This may return NaN for m2 == 0 and m3 == 0, which is expected.
     if (biased) {
-      result = sqrt(count) * m3 / sqrt(m2 * m2 * m2);
+      result = sqrt((double)count) * m3 / sqrt(m2 * m2 * m2);
     } else {
       auto m2_avg = m2 / count;
-      result = sqrt(count * (count - 1)) / (count - 2) * (m3 / count) /
+      result = sqrt((double)(count * (count - 1))) / (count - 2) * (m3 / count) /
                sqrt(m2_avg * m2_avg * m2_avg);
     }
     return result;
--- a/cpp/src/arrow/compare.cc	Mon Oct 20 15:48:39 2025
+++ b/cpp/src/arrow/compare.cc	Wed Dec 17 12:11:03 2025
@@ -111,7 +111,7 @@
     Float16 f_x = Float16::FromBits(x);
     Float16 f_y = Float16::FromBits(y);
     if (f_x == f_y) {
-      return Flags::signed_zeros_equal || (f_x.signbit() == f_y.signbit());
+      return Flags::signed_zeros_equal || (f_x.arsignbit() == f_y.arsignbit());
     }
     if (Flags::nans_equal && f_x.is_nan() && f_y.is_nan()) {
       return true;
--- a/cpp/src/arrow/compute/kernels/base_arithmetic_internal.h	Mon Oct 20 15:48:39 2025
+++ b/cpp/src/arrow/compute/kernels/base_arithmetic_internal.h	Wed Dec 17 12:16:59 2025
@@ -653,7 +653,7 @@
     return arg.is_nan()
                ? arg
                : (arg.is_zero() ? Float16::zero()
-                                : (arg.signbit() ? -Float16::one() : Float16::one()));
+                                : (arg.arsignbit() ? -Float16::one() : Float16::one()));
   }
 
   template <typename T, typename Arg>
