--- caja-1.22.3/libcaja-private/caja-progress-info.c~	Mon Dec 23 17:17:34 2019
+++ caja-1.22.3/libcaja-private/caja-progress-info.c	Sat Apr 18 16:24:00 2020
@@ -621,9 +621,16 @@
             {
                 NotifyNotification *notification;
 
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
                 notification = notify_notification_new ("caja",
                                                         _("Process completed"),
                                                         "system-file-manager");
+#else
+                notification = notify_notification_new ("caja",
+                                                        _("Process completed"),
+                                                        "system-file-manager",
+							NULL);
+#endif
 
                 notify_notification_show (notification, NULL);
 
