--- caja-1.22.3/libcaja-private/caja-file-operations.c~	Mon Dec 23 17:17:34 2019
+++ caja-1.22.3/libcaja-private/caja-file-operations.c	Sat Apr 18 16:28:54 2020
@@ -207,11 +207,15 @@
 
     if (unmount_notify == NULL) {
         unmount_notify =
-                        notify_notification_new (strings[0], strings[1],
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+		notify_notification_new (strings[0], strings[1],
                                                  "media-removable");
-
         notify_notification_set_hint (unmount_notify,
                                       "transient", g_variant_new_boolean (TRUE));
+#else
+		notify_notification_new (strings[0], strings[1],
+					 "media-removable", NULL);
+#endif
         notify_notification_set_urgency (unmount_notify,
                                          NOTIFY_URGENCY_CRITICAL);
     } else {
