--- cups-pdf.h~	Fri Feb 24 15:31:00 2017
+++ cups-pdf.h	Wed Oct 11 16:29:50 2023
@@ -58,7 +58,7 @@
   { "AnonDirName", SEC_CONF|SEC_PPD, { "/var/spool/cups-pdf/ANONYMOUS" } },
   { "AnonUser", SEC_CONF|SEC_PPD, { "nobody" } },
   { "GhostScript", SEC_CONF|SEC_PPD, { "/usr/bin/gs" } },
-  { "GSCall", SEC_CONF|SEC_PPD, { "%s -q -dCompatibilityLevel=%s -dNOPAUSE -dBATCH -dSAFER -sDEVICE=pdfwrite -sOutputFile=\"%s\" -dAutoRotatePages=/PageByPage -dAutoFilterColorImages=false -dColorImageFilter=/FlateEncode -dPDFSETTINGS=/prepress -c .setpdfwrite -f %s" } },
+  { "GSCall", SEC_CONF|SEC_PPD, { "%s -q -dCompatibilityLevel=%s -dNOPAUSE -dBATCH -dSAFER -sDEVICE=pdfwrite -sOutputFile=\"%s\" -dAutoRotatePages=/PageByPage -dAutoFilterColorImages=false -dColorImageFilter=/FlateEncode -dPDFSETTINGS=/prepress -f %s" } },
   { "Grp", SEC_CONF|SEC_PPD, { "lp" } },
   { "GSTmp", SEC_CONF|SEC_PPD, { "TMPDIR=/var/tmp" } },
   { "Log", SEC_CONF|SEC_PPD, { "/var/log/cups" } },
