--- feh-3.8/config.mk~	Mon Jan  3 10:34:57 2022
+++ feh-3.8/config.mk	Fri Jan 14 18:14:12 2022
@@ -1,18 +1,20 @@
 PACKAGE ?= feh
 VERSION ?= 3.10.3
 
+CC=gcc
+
 app ?= 0
 curl ?= 1
 debug ?= 0
-exif ?= 0
+exif ?= 1
 help ?= 0
 magic ?= 0
 mkstemps ?= 1
-verscmp ?= 1
+verscmp ?= 0
 xinerama ?= 1
 
 # Prefix for all installed files
-PREFIX ?= /usr/local
+PREFIX ?= /usr
 ICON_PREFIX ?= ${DESTDIR}${PREFIX}/share/icons
 
 # icons in /usr/share/local/icons (and other prefixes != /usr) are not
@@ -40,7 +42,7 @@
 CFLAGS += -Wall -Wextra -pedantic
 
 # Settings for glibc >= 2.19 - may need to be adjusted for other systems
-CFLAGS += -std=c11 -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_DARWIN_C_SOURCE
+CFLAGS += -std=c11 -m64 -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D__EXTENSIONS__
 
 ifeq (${curl},1)
 	CFLAGS += -DHAVE_LIBCURL
