--- gobject-introspection-1.44.0/giscanner/ccompiler.py~	Wed Sep  3 07:03:58 2014
+++ gobject-introspection-1.44.0/giscanner/ccompiler.py	Sat Apr 18 12:09:28 2020
@@ -73,7 +73,7 @@
 
         # Ensure libraries are always linked as we are going to use ldd to work
         # out their names later
-        if not libtool and pkgconfig_msvc_flags == '':
+        if not libtool and pkgconfig_msvc_flags == '' and sys.platform != 'sunos5':
             args.append('-Wl,--no-as-needed')
 
         for library in libraries:
