--- a/giscanner/ccompiler.py	Fri Mar 18 11:53:54 2022
+++ a/giscanner/ccompiler.py	Mon Apr  3 18:00:59 2023
@@ -197,7 +197,7 @@
 
                 # Ensure libraries are always linked as we are going to use ldd to work
                 # out their names later
-                if sys.platform != 'darwin':
+                if sys.platform != 'darwin' and sys.platform != 'sunos5':
                     args.append('-Wl,--no-as-needed')
 
         for library_path in libpaths:
