--- a/giscanner/shlibs.py	Tue Aug  5 15:34:40 2014
+++ b/giscanner/shlibs.py	Fri Apr  7 19:05:20 2023
@@ -118,11 +118,6 @@
                     shlibs.append(m.group(1))
                     break
 
-        if len(patterns) > 0:
-            raise SystemExit(
-                "ERROR: can't resolve libraries to shared libraries: " +
-                ", ".join(patterns.keys()))
-
     return shlibs
 
 
