*** harfbuzz-0.9.26/configure	Thu Jan 30 10:12:16 2014
--- b/harfbuzz-0.9.26/configure	Sat Feb  1 05:03:24 2014
***************
*** 16972,16986 ****
  	CXXFLAGS="$CXXFLAGS -Wcast-align"
  
  	case "$host" in
- 		*-*-mingw*)
- 		;;
- 		*)
- 			# Hide inline methods
- 			CXXFLAGS="$CXXFLAGS -fvisibility-inlines-hidden"
- 		;;
- 	esac
- 
- 	case "$host" in
  		arm-*-*)
  			# Request byte alignment on arm
  			CXXFLAGS="$CXXFLAGS -mstructure-size-boundary=8"
--- 16972,16977 ----
