--- a/src/2geom/math-utils.h	Sun Nov 30 18:45:32 2014
+++ b/src/2geom/math-utils.h	Sun Jul 12 18:14:06 2015
@@ -40,6 +40,9 @@
 #include <math.h> // sincos is usually only available in math.h
 #include <cmath>
 #include <utility> // for std::pair
+#if defined (SOLARIS)
+#include  <ieeefp.h>
+#endif
 #include <boost/math/special_functions/fpclassify.hpp>
 
 namespace Geom {
@@ -111,7 +114,7 @@
 # define IS_NAN(_a) (_isnan(_a)) 	/* Win32 definition */
 #elif defined(isnan) || defined(__FreeBSD__) || defined(__osf__)
 # define IS_NAN(_a) (isnan(_a))		/* GNU definition */
-#elif defined (SOLARIS)
+#elif defined (SOLARIS_XX)
 # define IS_NAN(_a) (isnan(_a))		/* GNU definition */
 #else
 # define IS_NAN(_a) (boost::math::isnan(_a))
