--- a/src/certificatehelper.cpp	Sun Jan 30 15:55:41 2022
+++ b/src/certificatehelper.cpp	Wed Apr 19 19:38:48 2023
@@ -18,7 +18,10 @@
 
     EVP_PKEY * pkey = EVP_PKEY_new();
     q_check_ptr(pkey);
-    RSA * rsa = RSA_generate_key(2048, RSA_F4, nullptr, nullptr);
+    RSA * rsa = RSA_new();
+    BIGNUM *bne = BN_new();
+    BN_set_word(bne, RSA_F4);
+    RSA_generate_key_ex(rsa, 2048, bne, nullptr);
     q_check_ptr(rsa);
     EVP_PKEY_assign_RSA(pkey, rsa);
     X509 * x509 = X509_new();
--- a/doc/gem-to-man.awk	Sun Jan 30 15:55:41 2022
+++ b/doc/gem-to-man.awk	Wed Apr 19 20:01:14 2023
@@ -1,4 +1,4 @@
-#!/usr/bin/env -S awk -f
+#!/usr/gnu/bin/gawk -f
 
 BEGIN {
     preformatted=0
