--- a/cupsfilters/ghostscript.c	Tue Jun 20 13:36:35 2023
+++ b/cupsfilters/ghostscript.c	Sat Sep 16 14:12:38 2023
@@ -453,6 +453,7 @@
   char *msg;
   int status = 65536;
   int wstatus;
+  extern char **environ;
 
   // Put Ghostscript command line argument into an array for the "exec()"
   // call
@@ -610,7 +611,8 @@
     }
 
     // Execute Ghostscript command line ...
-    execvpe(filename, gsargv, envp);
+    environ = envp;
+    execvp(filename, gsargv);
     if (log) log(ld, CF_LOGLEVEL_ERROR,
 		 "cfFilterGhostscript: Unable to launch Ghostscript: %s: %s",
 		 filename, strerror(errno));
