--- libgphoto2-2.5.14/libgphoto2_port/libgphoto2_port/libgphoto2_port.ver.1	2017-07-26 13:36:57.281790508 +0000
+++ libgphoto2-2.5.14/libgphoto2_port/libgphoto2_port/libgphoto2_port.ver	2017-07-26 13:43:49.329465627 +0000
@@ -71,7 +71,6 @@
 
 # These are only supposed to be used by libgphoto2 internally.
 LIBGPHOTO2_INTERNAL {
-	gpi_gphoto_port_type_map;
 	gpi_enum_to_string;
 	gpi_string_to_enum;
 	gpi_string_to_flag;
