*** a/libreoffice-4.1.4.2/sal/osl/unx/file.cxx        Tue Dec 10 08:11:21 2013
--- b/libreoffice-4.1.4.2/sal/osl/unx/file.cxx  Fri Jan 17 13:51:26 2014
***************
*** 1231,1237 ****
              OSL_TRACE(
                  "posix_madvise(..., POSIX_MADV_WILLNEED) failed with %d", e);
          }
! #elif defined SOLARIS
          if (madvise(static_cast< caddr_t >(p), nLength, MADV_WILLNEED) != 0)
          {
              OSL_TRACE("madvise(..., MADV_WILLNEED) failed with %d", errno);
--- 1231,1237 ----
              OSL_TRACE(
                  "posix_madvise(..., POSIX_MADV_WILLNEED) failed with %d", e);
          }
! #elif defined SOLARISNOTREALLY
          if (madvise(static_cast< caddr_t >(p), nLength, MADV_WILLNEED) != 0)
          {
              OSL_TRACE("madvise(..., MADV_WILLNEED) failed with %d", errno);
