*** a/extensions/Executable_pluginapp.bin.mk	Wed Feb 26 12:40:16 2014
--- b/extensions/Executable_pluginapp.bin.mk	Tue Mar 25 07:04:18 2014
***************
*** 72,84 ****
      gtk \
  ))
  
- # the orignal dmakefile said: don't ask, it's ugly
- ifeq ($(OS),SOLARIS)
- $(eval $(call gb_Executable_set_ldflags,pluginapp.bin,\
- 	-z nodefs \
- ))
  endif
  
- endif
- 
  # vim:set noet sw=4 ts=4:
--- 72,77 ----
