diff -wpruN '--exclude=*.orig' a~/nspr/configure a/nspr/configure
--- a~/nspr/configure	1970-01-01 00:00:00
+++ a/nspr/configure	1970-01-01 00:00:00
@@ -8069,7 +8069,7 @@ fi
     # be linked against -lrt (or -lposix4) so add it to OS_LIBS
     _rev=`uname -r`
     _librt=`echo $_rev 5.6 | awk '{ if ($1 > $2) print "-lrt"; else print "-lposix4" }'`
-    OS_LIBS="$OS_LIBS $_librt"
+    # OS_LIBS="$OS_LIBS $_librt"
     ;;
 
 *)
diff -wpruN '--exclude=*.orig' a~/nss/coreconf/SunOS5.mk a/nss/coreconf/SunOS5.mk
--- a~/nss/coreconf/SunOS5.mk	1970-01-01 00:00:00
+++ a/nss/coreconf/SunOS5.mk	1970-01-01 00:00:00
@@ -139,4 +139,4 @@ else
 RPATH = -R '$$ORIGIN'
 endif
 
-OS_LIBS += -lthread -lnsl -lsocket -lposix4 -ldl -lc
+OS_LIBS += -lnsl -lsocket -ldl -lc
