diff -wpruN '--exclude=*.orig' a~/nss/lib/freebl/verified/kremlib.h a/nss/lib/freebl/verified/kremlib.h
--- a~/nss/lib/freebl/verified/kremlib.h	1970-01-01 00:00:00
+++ a/nss/lib/freebl/verified/kremlib.h	1970-01-01 00:00:00
@@ -205,6 +205,7 @@ typedef const char *Prims_string;
 
 /* ... for Solaris */
 #elif defined(__sun__)
+#include <endian.h>
 #include <sys/byteorder.h>
 #define htole64(x) LE_64(x)
 #define le64toh(x) LE_64(x)
