diff -wpruN '--exclude=*.orig' a~/nss/Makefile a/nss/Makefile
--- a~/nss/Makefile	1970-01-01 00:00:00
+++ a/nss/Makefile	1970-01-01 00:00:00
@@ -96,6 +96,7 @@ endif
 NSPR_CONFIGURE_ENV := $(filter-out -arch x86_64,$(NSPR_CONFIGURE_ENV))
 NSPR_CONFIGURE_ENV := $(filter-out -arch i386,$(NSPR_CONFIGURE_ENV))
 NSPR_CONFIGURE_ENV := $(filter-out -arch ppc,$(NSPR_CONFIGURE_ENV))
+NSPR_CONFIGURE_ENV := $(filter-out -Wall -Wno-format,$(NSPR_CONFIGURE_ENV))
 
 #
 # Some pwd commands on Windows (for example, the pwd
