diff --git a/palemoon/installer/package-manifest.in b/palemoon/installer/package-manifest.in
index c46831a1c..eff4ee887 100644
--- a/palemoon/installer/package-manifest.in
+++ b/palemoon/installer/package-manifest.in
@@ -74,9 +74,6 @@
 #ifdef GKMEDIAS_SHARED_LIBRARY
 @BINPATH@/@DLL_PREFIX@gkmedias@DLL_SUFFIX@
 #endif
-#ifndef MOZ_SYSTEM_HUNSPELL
-@BINPATH@/@DLL_PREFIX@hunspell@DLL_SUFFIX@
-#endif
 #ifndef MOZ_SYSTEM_NSPR
 #ifndef MOZ_FOLD_LIBS
 @BINPATH@/@DLL_PREFIX@nspr4@DLL_SUFFIX@
