--- a/hacks/glx/papercube.c	Wed Aug  2 19:45:42 2023
+++ b/hacks/glx/papercube.c	Thu Oct 26 15:08:16 2023
@@ -858,7 +858,7 @@
   double angle = 90.0;
   double fold = FOLD_DURATION / speed;
   double pause = PAUSE_DURATION / speed;
-  double sun = SUN_DURATION / speed;
+  double xxxsun = SUN_DURATION / speed;
   double spin = SPIN_DURATION / speed;
   double spin_rps = SPIN_RPS;
   double time = 0;
@@ -866,7 +866,7 @@
   struct field_move *field_move;
   int i;
 
-  set_move(&moves->sunrise, time, sun, 0.0, brightness);
+  set_move(&moves->sunrise, time, xxxsun, 0.0, brightness);
 
   time = moves->sunrise.stop + pause;
 
@@ -901,12 +901,12 @@
 
   time = field_move->move.stop + pause;
 
-  set_move(&moves->spin, time, time + spin + sun,
+  set_move(&moves->spin, time, time + spin + xxxsun,
            0.0, (spin + sun) * spin_rps * 360);
 
   time += spin;
 
-  set_move(&moves->sunset, time, time + sun, brightness, 0.0);
+  set_move(&moves->sunset, time, time + xxxsun, brightness, 0.0);
 
   moves->stage = Sunrise;
   set_move_time_big_bang(&moves->move_time);
