--- a/qtdeclarative/src/qml/jsruntime/qv4globalobject.cpp
+++ b/qtdeclarative/src/qml/jsruntime/qv4globalobject.cpp
@@ -534,7 +534,7 @@
 ReturnedValue GlobalFunctions::method_parseInt(CallContext *ctx)
     }

     if (overflow) {
-        double result = (double) v_overflow * pow(R, overflow_digit_count);
+        double result = (double) v_overflow * pow(double(R), int(overflow_digit_count));
         result += v;
         return Encode(sign * result);
     } else {
