diff -wpruN '--exclude=*.orig' a~/fdname.c a/fdname.c
--- a~/fdname.c	1970-01-01 00:00:00
+++ a/fdname.c	1970-01-01 00:00:00
@@ -261,7 +261,10 @@ int sockname(int fd, FILE *outfile, char
    }
    strncpy(protoname, protoentp->p_name, sizeof(protoname));
 #elif HAVE_GETPROTOBYNUMBER_R==2 /* Solaris */
-   protoentp = getprotobynumber(proto);
+   char buf[PROTONAMEMAX];
+   protoentp = getprotobynumber_r(proto, &protoent, buf, sizeof(buf));
+   if (protoentp == NULL)
+      Warn1("sockname(): getprotobynumber_r(proto=%d, ...)", proto);
    strncpy(protoname, protoentp->p_name, sizeof(protoname));
 #elif HAVE_GETPROTOBYNUMBER_R==3 /* AIX */
    {
