diff -wpruN '--exclude=*.orig' a~/fdname.c a/fdname.c
--- a~/fdname.c	1970-01-01 00:00:00
+++ a/fdname.c	1970-01-01 00:00:00
@@ -91,7 +91,11 @@ static int procgetfdname(int fd, char *f
 #endif /* !HAVE_STAT64 */
        
    if (pid < 0)  pid = Getpid();
+#ifdef __illumos__
+   snprintf(procpath, sizeof(procpath), "/proc/"F_pid"/path/%d", pid, fd);
+#else
    snprintf(procpath, sizeof(procpath), "/proc/"F_pid"/fd/%d", pid, fd);
+#endif
    if ((len = Readlink(procpath, filepath, pathsize-1)) < 0) {
       Error4("readlink(\"%s\", %p, "F_Zu"): %s",
 	     procpath, filepath, pathsize, strerror(errno));
