--- a/src/sthreads.c	Mon Nov 30 20:40:04 2020
+++ b/src/sthreads.c	Mon Nov 30 20:50:26 2020
@@ -112,7 +112,7 @@
 #if OPENSSL_VERSION_NUMBER>=0x10000000L && OPENSSL_VERSION_NUMBER<0x10100000L
     CRYPTO_THREADID_set_callback(threadid_func);
 #endif
-#if OPENSSL_VERSION_NUMBER<0x10000000L || !defined(OPENSSL_NO_DEPRECATED)
+#if OPENSSL_VERSION_NUMBER<0x10000000L
     CRYPTO_set_id_callback(stunnel_thread_id);
 #endif
 }
