--- a/tools/winebuild/utils.c	Sat Dec 19 15:57:44 2015
+++ b/tools/winebuild/utils.c	Mon Dec 21 19:27:53 2015
@@ -447,6 +447,9 @@
         case PLATFORM_FREEBSD:
             strarray_add( &args, "-m", (force_pointer_size == 8) ? "elf_x86_64_fbsd" : "elf_i386_fbsd", NULL );
             break;
+        case PLATFORM_SOLARIS:
+            strarray_add( &args, "-m", (force_pointer_size == 8) ? "elf_x86_64_sol2" : "elf_i386_sol2", NULL );
+            break;
         default:
             switch(target_cpu)
             {
