--- libreoffice-7.2.4.1/sal/osl/unx/socket.cxx.1	2020-03-14 13:10:11.725559641 +0000
+++ libreoffice-7.2.4.1/sal/osl/unx/socket.cxx	2020-03-14 13:10:54.911745912 +0000
@@ -864,7 +864,7 @@
     static auto const init = [bUseFQDN]() -> std::pair<oslSocketResult, OUString> {
             char LocalHostname[256] = "";
 
-#ifdef SYSV
+#if defined(SYSV) && !defined(__sun__)
             struct utsname uts;
 
             if (uname(&uts) < 0)
