--- libreoffice-7.2.4.1/tools/source/datetime/ttime.cxx~	Fri Dec  3 00:10:29 2021
+++ libreoffice-7.2.4.1/tools/source/datetime/ttime.cxx	Sun Jan  2 21:15:38 2022
@@ -415,7 +415,7 @@
 #if defined(__sun)
         // Solaris gmtime_r() seems not to handle daylight saving time
         // flags correctly
-        nUTC = nLocalTime + ( aTM.tm_isdst == 0 ? timezone : altzone );
+        sal_Int32 nUTC = nLocalTime + ( aTM.tm_isdst == 0 ? timezone : altzone );
 #elif defined( LINUX )
         // Linux mktime() seems not to handle tm_isdst correctly
         sal_Int32 nUTC = nLocalTime - aTM.tm_gmtoff;
