diff -wpruN '--exclude=*.orig' a~/jdk/make/CompileLaunchers.gmk a/jdk/make/CompileLaunchers.gmk
--- a~/jdk/make/CompileLaunchers.gmk	1970-01-01 00:00:00
+++ a/jdk/make/CompileLaunchers.gmk	1970-01-01 00:00:00
@@ -163,7 +163,7 @@ define SetupLauncher
           -DLAUNCHER_NAME='"$(LAUNCHER_NAME)"' \
           -DPROGNAME='"$1"' $(DPACKAGEPATH) \
           $2, \
-      CFLAGS_solaris := -KPIC, \
+      CFLAGS_solaris := -fPIC, \
       LDFLAGS := $(LDFLAGS_JDKEXE) \
           $(ORIGIN_ARG) \
           $$($1_LDFLAGS), \
@@ -516,7 +516,7 @@ $(eval $(call SetupNativeCompilation,BUI
         -DFULL, \
     CFLAGS_release := -DPRODUCT, \
     CFLAGS_linux := -fPIC, \
-    CFLAGS_solaris := -KPIC, \
+    CFLAGS_solaris := -fPIC, \
     CFLAGS_macosx := -fPIC, \
     MAPFILE := $(UNPACK_MAPFILE),\
     LDFLAGS := $(UNPACKEXE_ZIPOBJS), \
@@ -626,7 +626,7 @@ ifneq ($(BUILD_JEXEC_SRC), )
       CFLAGS := $(CFLAGS_JDKEXE) \
           $(BUILD_JEXEC_INC), \
       CFLAGS_linux := -fPIC, \
-      CFLAGS_solaris := -KPIC, \
+      CFLAGS_solaris := -fPIC, \
       LDFLAGS := $(LDFLAGS_JDKEXE) \
           $(call SET_SHARED_LIBRARY_NAME,$(LIBRARY_PREFIX)$(SHARED_LIBRARY_SUFFIX)), \
       OBJECT_DIR := $(JDK_OUTPUTDIR)/objs/jexec_obj, \
