diff -wpruN '--exclude=*.orig' a~/common/autoconf/generated-configure.sh a/common/autoconf/generated-configure.sh
--- a~/common/autoconf/generated-configure.sh	1970-01-01 00:00:00
+++ a/common/autoconf/generated-configure.sh	1970-01-01 00:00:00
@@ -43495,7 +43495,7 @@ $as_echo "$supports" >&6; }
       # Enabling pie on 32 bit builds prevents the JVM from allocating a continuous
       # java heap.
       if test "x$OPENJDK_TARGET_CPU_BITS" != "x32"; then
-        LDFLAGS_JDKEXE="$LDFLAGS_JDKEXE -pie"
+        LDFLAGS_JDKEXE="$LDFLAGS_JDKEXE"
       fi
     fi
   fi
