diff -wpruN '--exclude=*.orig' a~/jdk/make/lib/Awt2dLibraries.gmk a/jdk/make/lib/Awt2dLibraries.gmk
--- a~/jdk/make/lib/Awt2dLibraries.gmk	1970-01-01 00:00:00
+++ a/jdk/make/lib/Awt2dLibraries.gmk	1970-01-01 00:00:00
@@ -677,7 +677,6 @@ $(eval $(call SetupNativeCompilation,BUI
         $(SHARED_LIBRARY_FLAGS) \
         -I$(JDK_TOPDIR)/src/share/native/sun/java2d \
         -I$(JDK_TOPDIR)/src/share/native/sun/awt/debug, \
-    CFLAGS_solaris := -xc99=no_lib, \
     CFLAGS_windows := -DCMS_IS_WINDOWS_, \
     MAPFILE := $(JDK_TOPDIR)/make/mapfiles/liblcms/mapfile-vers, \
     LDFLAGS := $(LDFLAGS_JDKLIB) \
@@ -1101,7 +1100,7 @@ else # OPENJDK_TARGET_OS not windows
       JAWT_LIBS += -lawt
     endif
     ifndef BUILD_HEADLESS_ONLY
-      JAWT_LIBS += -lawt_xawt
+      JAWT_LIBS += $(X_LIBS) -lawt_xawt
     else
       JAWT_LIBS += -lawt_headless
       HEADLESS_CFLAG += -DHEADLESS
@@ -1125,7 +1124,7 @@ else # OPENJDK_TARGET_OS not windows
       LDFLAGS_solaris := $(X_LIBS), \
       LDFLAGS_SUFFIX_linux := $(JAWT_LIBS) $(LDFLAGS_JDKLIB_SUFFIX), \
       LDFLAGS_SUFFIX_aix := $(JAWT_LIBS) $(LDFLAGS_JDKLIB_SUFFIX),\
-      LDFLAGS_SUFFIX_solaris := $(JAWT_LIBS) $(LDFLAGS_JDKLIB_SUFFIX) -lXrender, \
+      LDFLAGS_SUFFIX_solaris := $(JAWT_LIBS) $(LDFLAGS_JDKLIB_SUFFIX), \
       LDFLAGS_SUFFIX_macosx := -Xlinker -rpath -Xlinker @loader_path $(JAWT_LIBS) \
           -framework Cocoa $(LDFLAGS_JDKLIB_SUFFIX), \
       OBJECT_DIR := $(JDK_OUTPUTDIR)/objs/libjawt, \
