$NetBSD: patch-jdk_make_CompileLaunchers.gmk,v 1.3 2016/05/06 13:37:01 ryoon Exp $

diff -wpruN '--exclude=*.orig' a~/jdk/make/CompileLaunchers.gmk a/jdk/make/CompileLaunchers.gmk
--- a~/jdk/make/CompileLaunchers.gmk	1970-01-01 00:00:00
+++ a/jdk/make/CompileLaunchers.gmk	1970-01-01 00:00:00
@@ -593,10 +593,8 @@ BUILD_JEXEC_DST_DIR := $(JDK_OUTPUTDIR)/
 # jdk/make/java/Makefile
 #
 ifeq ($(OPENJDK_TARGET_OS), solaris)
-  ifeq ($(OPENJDK_TARGET_CPU_BITS), 32)
     BUILD_JEXEC := 1
   endif
-endif
 
 ifeq ($(OPENJDK_TARGET_OS), linux)
   BUILD_JEXEC := 1
