$NetBSD: patch-common_autoconf_spec.gmk.in,v 1.4 2015/06/10 13:29:57 tnn Exp $

diff -wpruN '--exclude=*.orig' a~/common/autoconf/spec.gmk.in a/common/autoconf/spec.gmk.in
--- a~/common/autoconf/spec.gmk.in	1970-01-01 00:00:00
+++ a/common/autoconf/spec.gmk.in	1970-01-01 00:00:00
@@ -392,7 +392,8 @@ endif
 USING_BROKEN_SUSE_LD:=@USING_BROKEN_SUSE_LD@
 
 # LDFLAGS used to link the jdk native libraries (C-code)
-LDFLAGS_JDKLIB:=@LDFLAGS_JDKLIB@
+# Do not use -Wl,-R here. The build system chokes on "," in LDFLAGS. -Xlinker works.
+LDFLAGS_JDKLIB:=-Xlinker -R\$$$$ORIGIN/server  @LDFLAGS_JDKLIB@
 LDFLAGS_JDKLIB_SUFFIX:=@LDFLAGS_JDKLIB_SUFFIX@
 
 # On some platforms the linker cannot be used to create executables, thus
@@ -400,7 +401,7 @@ LDFLAGS_JDKLIB_SUFFIX:=@LDFLAGS_JDKLIB_S
 LDEXE:=@FIXPATH@ @LDEXE@
 
 # LDFLAGS used to link the jdk native launchers (C-code)
-LDFLAGS_JDKEXE:=@LDFLAGS_JDKEXE@
+LDFLAGS_JDKEXE:=-Xlinker -R\$$$$ORIGIN/server @LDFLAGS_JDKEXE@
 LDFLAGS_JDKEXE_SUFFIX:=@LDFLAGS_JDKEXE_SUFFIX@
 
 # LDFLAGS specific to C++ linking.
@@ -490,7 +491,7 @@ JAVA_SMALL=@FIXPATH@ $(BOOT_JDK)/bin/jav
 JAVAC=@FIXPATH@ $(BOOT_JDK)/bin/javac
 # Hotspot sets this variable before reading the SPEC when compiling sa-jdi.jar. Avoid
 # overriding that value by using ?=.
-JAVAC_FLAGS?=@JAVAC_FLAGS@
+JAVAC_FLAGS?=@JAVAC_FLAGS@ -encoding UTF-8 -verbose
 
 JAVAH=@FIXPATH@ $(BOOT_JDK)/bin/javah
 
