diff -wpruN '--exclude=*.orig' a~/CPP/7zip/Archive/Wim/WimHandler.cpp a/CPP/7zip/Archive/Wim/WimHandler.cpp
--- a~/CPP/7zip/Archive/Wim/WimHandler.cpp	1970-01-01 00:00:00
+++ a/CPP/7zip/Archive/Wim/WimHandler.cpp	1970-01-01 00:00:00
@@ -298,14 +298,14 @@ STDMETHODIMP CHandler::GetArchivePropert
 
       AString res;
 
-      bool numMethods = 0;
+      bool numMethods = false;
       for (unsigned i = 0; i < ARRAY_SIZE(k_Methods); i++)
       {
         if (methodMask & ((UInt32)1 << i))
         {
           res.Add_Space_if_NotEmpty();
           res += k_Methods[i];
-          numMethods++;
+          numMethods = true;
         }
       }
 
@@ -315,10 +315,10 @@ STDMETHODIMP CHandler::GetArchivePropert
         ConvertUInt32ToString(methodUnknown, temp);
         res.Add_Space_if_NotEmpty();
         res += temp;
-        numMethods++;
+        numMethods = true;
       }
 
-      if (numMethods == 1 && chunkSizeBits != 0)
+      if (numMethods && chunkSizeBits != 0)
       {
         char temp[32];
         temp[0] = ':';
