https://sourceforge.net/p/p7zip/bugs/226/

diff -wpruN '--exclude=*.orig' a~/CPP/Windows/ErrorMsg.cpp a/CPP/Windows/ErrorMsg.cpp
--- a~/CPP/Windows/ErrorMsg.cpp	1970-01-01 00:00:00
+++ a/CPP/Windows/ErrorMsg.cpp	1970-01-01 00:00:00
@@ -13,7 +13,7 @@ UString MyFormatMessage(DWORD errorCode)
   const char * txt = 0;
   AString msg;
 
-  switch(errorCode) {
+  switch(HRESULT(errorCode)) {
     case ERROR_NO_MORE_FILES   : txt = "No more files"; break ;
     case E_NOTIMPL             : txt = "E_NOTIMPL"; break ;
     case E_NOINTERFACE         : txt = "E_NOINTERFACE"; break ;
@@ -43,7 +43,7 @@ bool MyFormatMessage(DWORD messageID, CS
   const char * txt = 0;
   AString msg;
 
-  switch(messageID) {
+  switch(HRESULT(messageID)) {
     case ERROR_NO_MORE_FILES   : txt = "No more files"; break ;
     case E_NOTIMPL             : txt = "E_NOTIMPL"; break ;
     case E_NOINTERFACE         : txt = "E_NOINTERFACE"; break ;
