diff -wpruN '--exclude=*.orig' a~/setup.py a/setup.py
--- a~/setup.py	1970-01-01 00:00:00
+++ a/setup.py	1970-01-01 00:00:00
@@ -1661,7 +1661,7 @@ class PyBuildExt(build_ext):
             self.missing.extend(['resource', 'termios'])
 
         # Platform-specific libraries
-        if HOST_PLATFORM.startswith(('linux', 'freebsd', 'gnukfreebsd')):
+        if HOST_PLATFORM.startswith(('sunos5', 'linux', 'freebsd', 'gnukfreebsd')):
             self.add(Extension('ossaudiodev', ['ossaudiodev.c']))
         elif HOST_PLATFORM.startswith(('netbsd')):
             self.add(Extension('ossaudiodev', ['ossaudiodev.c'],
