
To expose the CMSG_ macros and new recvmsg() semantics, the socket module
needs to be built in an XPG6 environment.

diff -wpruN '--exclude=*.orig' a~/setup.py a/setup.py
--- a~/setup.py	1970-01-01 00:00:00
+++ a/setup.py	1970-01-01 00:00:00
@@ -971,7 +971,13 @@ class PyBuildExt(build_ext):
         # grp(3)
         self.addext(Extension('grp', ['grpmodule.c']))
 
-        self.addext(Extension('_socket', ['socketmodule.c']))
+        kwargs = {}
+        if ILLUMOS:
+            kwargs['libraries'] = ['socket', 'nsl', 'resolv']
+            kwargs['extra_compile_args'] = ['-D_XOPEN_SOURCE=600',
+                '-D__EXTENSIONS__=1']
+        self.addext(Extension('_socket', ['socketmodule.c'], **kwargs))
+
         self.addext(Extension('spwd', ['spwdmodule.c']))
 
         # select(2); not on ancient System V
