diff -wpruN '--exclude=*.orig' a~/setup.py a/setup.py
--- a~/setup.py	1970-01-01 00:00:00
+++ a/setup.py	1970-01-01 00:00:00
@@ -83,6 +83,7 @@ MS_WINDOWS = (HOST_PLATFORM == 'win32')
 CYGWIN = (HOST_PLATFORM == 'cygwin')
 MACOS = (HOST_PLATFORM == 'darwin')
 AIX = (HOST_PLATFORM.startswith('aix'))
+ILLUMOS = (HOST_PLATFORM == 'sunos5')
 VXWORKS = ('vxworks' in HOST_PLATFORM)
 EMSCRIPTEN = HOST_PLATFORM == 'emscripten-wasm32'
 CC = os.environ.get("CC")
@@ -847,7 +848,7 @@ class PyBuildExt(build_ext):
         # Ensure that /usr/local is always used, but the local build
         # directories (i.e. '.' and 'Include') must be first.  See issue
         # 10520.
-        if not CROSS_COMPILING:
+        if not ILLUMOS and not CROSS_COMPILING:
             add_dir_to_list(self.compiler.library_dirs, '/usr/local/lib')
             add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
         # only change this for cross builds for 3.3, issues on Mageia
