diff -wpruN '--exclude=*.orig' a~/Tools/freeze/test/freeze.py a/Tools/freeze/test/freeze.py
--- a~/Tools/freeze/test/freeze.py	1970-01-01 00:00:00
+++ a/Tools/freeze/test/freeze.py	1970-01-01 00:00:00
@@ -10,7 +10,7 @@ TESTS_DIR = os.path.dirname(__file__)
 TOOL_ROOT = os.path.dirname(TESTS_DIR)
 SRCDIR = os.path.dirname(os.path.dirname(TOOL_ROOT))
 
-MAKE = shutil.which('make')
+MAKE = shutil.which('gmake')
 FREEZE = os.path.join(TOOL_ROOT, 'freeze.py')
 OUTDIR = os.path.join(TESTS_DIR, 'outdir')
 
