diff -wpruN --no-dereference '--exclude=*.orig' a~/agent/mibgroup/mibII.h a/agent/mibgroup/mibII.h
--- a~/agent/mibgroup/mibII.h	1970-01-01 00:00:00
+++ a/agent/mibgroup/mibII.h	1970-01-01 00:00:00
@@ -56,3 +56,10 @@ config_require(ip-mib/ipSystemStatsTable
  * ip-mib/ipv6ScopeZoneIndexTable
  */
 #endif
+
+/*
+ * For Solaris, enable additional tables.
+ */
+#if defined( solaris2 )
+config_require(ip-mib/ipSystemStatsTable ip-mib/ipAddressTable)
+#endif
