diff -wpruN --no-dereference '--exclude=*.orig' a~/agent/mibgroup/ucd-snmp/vmstat_solaris2.c a/agent/mibgroup/ucd-snmp/vmstat_solaris2.c
--- a~/agent/mibgroup/ucd-snmp/vmstat_solaris2.c	1970-01-01 00:00:00
+++ a/agent/mibgroup/ucd-snmp/vmstat_solaris2.c	1970-01-01 00:00:00
@@ -534,6 +534,13 @@ update_stats(unsigned int registrationNu
                     "time_diff: %lld\n", time_diff));
 
         /*
+		 * Assign a minimum value to get around divide by zero case
+		 */
+		if (time_diff == 0) {
+			time_diff = 1;
+		}
+
+        /*
          * swapin and swapout are in pages, MIB wants kB/s,so we just need to get kB and seconds 
          * For the others we need to get value per second 
          * getpagesize() returns pagesize in bytes 
