diff -wpruN --no-dereference '--exclude=*.orig' a~/agent/mibgroup/kernel_sunos5.c a/agent/mibgroup/kernel_sunos5.c
--- a~/agent/mibgroup/kernel_sunos5.c	1970-01-01 00:00:00
+++ a/agent/mibgroup/kernel_sunos5.c	1970-01-01 00:00:00
@@ -293,6 +293,7 @@ getKstatInt(const char *classname, const
     kid_t           kid;
     kstat_named_t  *named;
     int             ret = -1;        /* fail unless ... */
+    int             lookup_stat = 0;
 
     if (kstat_fd == 0) {
 	kstat_fd = kstat_open();
@@ -303,6 +304,65 @@ getKstatInt(const char *classname, const
     if ((ksc = kstat_fd) == NULL) {
 	goto Return;
     }
+
+    if ((ks = kstat_lookup(ksc, NETSNMP_REMOVE_CONST(char *, statname),
+         -1, "link")) != NULL)
+    {
+        kid = kstat_read(ksc, ks, NULL);
+        if (kid == -1) {
+            DEBUGMSGTL(("kernel_sunos5", "cannot read class %s stats %s\n",
+                    statname ? statname : "NULL", "link"));
+        }
+        else
+        {
+            named = kstat_data_lookup(ks, varname);
+            if (named == NULL) {
+	        DEBUGMSGTL(("kernel_sunos5", "no var %s for class %s stat %s\n",
+                        varname, statname ? statname : "NULL", "link"));
+            }
+            else {
+                    switch (named->data_type) {
+#ifdef KSTAT_DATA_INT32         /* Solaris 2.6 and up */
+                        case KSTAT_DATA_INT32:
+                            if(named->value.i32)
+                                lookup_stat = 1;
+	                    break;
+                        case KSTAT_DATA_UINT32:
+                            if(named->value.ui32)
+                                lookup_stat = 1;
+                            break;
+                        case KSTAT_DATA_INT64:
+                            if(named->value.i64)
+                                lookup_stat = 1;
+                            break;
+                        case KSTAT_DATA_UINT64:
+                            if(named->value.ui64)
+                                lookup_stat = 1;
+                            break;
+#else
+                        case KSTAT_DATA_LONG:
+                            if(named->value.l)
+                                lookup_stat = 1;
+                            break;
+                        case KSTAT_DATA_ULONG:
+                            if(named->value.ul)
+                                lookup_stat = 1;
+                            break;
+                        case KSTAT_DATA_LONGLONG:
+                            if(named->value.ll)
+                                lookup_stat = 1;
+                            break;
+                        case KSTAT_DATA_ULONGLONG:
+                            if(named->value.ull)
+                                lookup_stat = 1;
+                            break;
+#endif
+                    }
+            }
+        }
+    }
+if (!lookup_stat) {
+
     ks = kstat_lookup(ksc, NETSNMP_REMOVE_CONST(char *, classname),
                       -1, NETSNMP_REMOVE_CONST(char *, statname));
     if (ks == NULL) {
@@ -324,7 +384,7 @@ getKstatInt(const char *classname, const
 		statname ? statname : "NULL"));
 	goto Return;
     }
-
+}
     ret = 0;                /* maybe successful */
     switch (named->data_type) {
 #ifdef KSTAT_DATA_INT32         /* Solaris 2.6 and up */
@@ -444,9 +504,16 @@ getKstat(const char *statname, const cha
      */
     if ((ks = kstat_lookup(ksc, module_name, instance,
                            NETSNMP_REMOVE_CONST(char *, statname))) == NULL) {
+        /*
+         * Try to lookup the kstat module with
+         * modulename = statname and statname = "link".
+         */
+      if ((ks = kstat_lookup(ksc, NETSNMP_REMOVE_CONST(char *, statname),
+          0, "link")) == NULL) {
 	ret = -10;
 	goto Return;        /* kstat errors */
     }
+    }
 
     if (kstat_read(ksc, ks, NULL) <= 0) {
 	ret = -10;
