diff -wpruN --no-dereference '--exclude=*.orig' a~/agent/netsnmp_close_fds.c a/agent/netsnmp_close_fds.c
--- a~/agent/netsnmp_close_fds.c	1970-01-01 00:00:00
+++ a/agent/netsnmp_close_fds.c	1970-01-01 00:00:00
@@ -7,6 +7,7 @@
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
+#include <stdlib.h>
 
 #include <net-snmp/agent/netsnmp_close_fds.h>
 
@@ -23,6 +24,11 @@ void netsnmp_close_fds(int fd)
     if (fd < -1)
         fd = -1;
 
+#ifdef __illumos__
+    closefrom(fd + 1);
+    return;
+#else
+
 #ifdef __linux__
     if ((dir = opendir("/proc/self/fd"))) {
         while ((ent = readdir(dir))) {
@@ -54,5 +60,6 @@ void netsnmp_close_fds(int fd)
 
     for (i = largest_fd; i > fd && i >= 0; i--)
         close(i);
+#endif /* __illumos__ */
 #endif
 }
