--- a/src/hotspot/share/utilities/globalDefinitions.hpp	Thu May 28 00:44:10 2020
+++ b/src/hotspot/share/utilities/globalDefinitions.hpp	Wed May 20 19:29:11 2020
@@ -790,6 +790,15 @@
 };
 
 
+#define STACK_BIAS      0
+// V9 Sparc CPU's running in 64 Bit mode use a stack bias of 7ff
+// in order to extend the reach of the stack pointer.
+#if defined(SPARC) && defined(_LP64)
+#undef STACK_BIAS
+#define STACK_BIAS      0x7ff
+#endif
+
+
 // TosState describes the top-of-stack state before and after the execution of
 // a bytecode or method. The top-of-stack value may be cached in one or more CPU
 // registers. The TosState corresponds to the 'machine representation' of this cached
