--- make/common/modules/LauncherCommon.gmk~	Thu Apr  9 18:18:09 2020
+++ make/common/modules/LauncherCommon.gmk	Tue Nov 26 00:55:54 2019
@@ -156,7 +156,7 @@
           -DLAUNCHER_NAME='"$$(LAUNCHER_NAME)"' \
           -DPROGNAME='"$1"' \
           $$($1_CFLAGS), \
-      CFLAGS_solaris := -KPIC, \
+      CFLAGS_solaris := -fPIC, \
       CFLAGS_windows := $$($1_CFLAGS_windows), \
       DISABLED_WARNINGS_gcc := unused-function, \
       LDFLAGS := $$(LDFLAGS_JDKEXE) \
