--- a/make/Main.gmk	Wed May 13 15:18:16 2020
+++ b/make/Main.gmk	Thu May 14 11:00:26 2020
@@ -1102,8 +1102,12 @@
 docs-image: docs-jdk
 
 # If not already set, set the JVM specific targets to build the test image
-JVM_TEST_IMAGE_TARGETS ?= test-image-hotspot-jtreg-native test-image-hotspot-gtest
+JVM_TEST_IMAGE_TARGETS ?= test-image-hotspot-jtreg-native
 
+ifeq ($(BUILD_GTEST), true)
+  JVM_TEST_IMAGE_TARGETS += test-image-hotspot-gtest
+endif
+
 ifeq ($(INCLUDE_GRAAL), true)
   JVM_TEST_IMAGE_TARGETS += test-image-hotspot-jtreg-graal
 endif
